components/wrong-network-modal.tsx

```diff
@@ -25,30 +26,74 @@ export default function WrongNetworkModal({requiredNetwork = ``}) {
+
   function getColumnClass() {
     const color = getColor();
 
     return `col-6 rounded border border-${color} text-${color} p-3 d-flex justify-content-between align-items-center`;
   }
+  function connectGithub(){
+    if(showModal())
+      localStorage.setItem(`lastAddressBeforeConnect`, currentAddress);
+      return signIn('github', {callbackUrl: `${window.location.protocol}//${window.location.host}/connect-account`})
+  }
 
-  return <Modal title="Change network" show={showModal()}>
-    <div className="text-center">
-      <strong className="capition d-block text-uppercase text-white-50 mb-4">
-        please, connect to the <span className="text-purple">kovan test network</span> on your metamask wallet
-      </strong>
-      <div className="row">
-        <div className="col-12 d-flex justify-content-center">
-          <div className={getColumnClass()}>
-            <div className="d-flex justify-content-start align-items-center">
-              <Image src={metamaskLogo} width={15} height={15}/> <span className="ms-2">{truncateAddress(currentAddress)}</span>
-            </div>
-            {!showModal() ? <CheckMarkIcon /> : <ErrorMarkIcon/>}
+  return (
+    <Modal
+      title="Connect your MetaMask Wallet"
+      titlePosition="center"
+      titleClass="h4 text-white bg-opacity-100"
+      show={showModal()}
+    >
+      <div className="text-center">
+        <strong className="smallCaption d-block text-uppercase text-white-50 mb-4">
+          to access this page please, connect to the{" "}
+          <span className="text-purple">kovan test network</span> on your
+          metamask wallet
+        </strong>
+        <div className="row">
+          <div className="col-12 d-flex justify-content-center">
+            {currentAddress ? (
+              <div className={getColumnClass()}>
+                <div className="d-flex justify-content-start align-items-center">
+                  <Image src={metamaskLogo} width={15} height={15} />{" "}
+                  <span className="ms-2">
+                    {truncateAddress(currentAddress)}
+                  </span>
+                </div>
+                {!showModal() ? <CheckMarkIcon /> : <ErrorMarkIcon />}
+              </div>
+            ) : (
+              <div className={`button-connect border bg-dark border-dark rounded d-flex justify-content-between p-3 align-items-center w-75 ${showModal() && `cursor-pointer`}`} onClick={connectGithub}>
+                <div className="mx-auto d-flex align-items-center text-uppercase smallCaption">
+                  <Image src={metamaskLogo} width={15} height={15} />{" "}
+                  <span className="ms-2 text-white text-opacity-1">
+                    metamask
+                  </span>
+                </div>
+              </div>
+            )}
           </div>
         </div>
+        <div className="smallInfo text-ligth-gray text-center fs-smallest text-dark text-uppercase mt-4">
+          BY CONNECTING, YOU{" "}
```

Comment:
you dont need to write uppercased letters, `text-uppercase` should be taking care of that (the class that this div has). 

You're also missing "agree" -- the text reads "by connecting, you Terms & Conditions"